assets/differentialabundance_report.Rmd

```diff
@@ -838,32 +880,67 @@ for (i in 1:nrow(contrasts)){
-          ref_gsea_results <- read_metadata(reference_gsea_tables[i])[,c(-2,-3)]
-          print( htmltools::tagList(datatable(ref_gsea_results, caption = paste0("\nReference (", contrasts$reference[i], ")\n"), rownames = FALSE) ))
+      if (gene_set_method == 'gsea') {
+        for (gmt_file in simpleSplit(params$mods_gene_sets)) {
+          gmt_name <- basename(tools::file_path_sans_ext(gmt_file))
+          cat("\n##### ", gmt_name ," {.tabset}\n")
+
+          reference_gsea_tables <- paste0(contrasts$id, ".", gmt_name, '.gsea_report_for_', contrasts$reference, '.tsv')
+          target_gsea_tables <- paste0(contrasts$id, ".", gmt_name, '.gsea_report_for_', contrasts$target, '.tsv')
+          for (i in 1:nrow(contrasts)){
+            cat("\n###### ", contrast_descriptions[i], "\n")
+            target_gsea_results <- read_metadata(target_gsea_tables[i])[,c(-2,-3)]
+            print( htmltools::tagList(datatable(target_gsea_results, caption = paste0("\nTarget (", contrasts$target[i], ")\n"), rownames = FALSE) ))
+            ref_gsea_results <- read_metadata(reference_gsea_tables[i])[,c(-2,-3)]
+            print( htmltools::tagList(datatable(ref_gsea_results, caption = paste0("\nReference (", contrasts$reference[i], ")\n"), rownames = FALSE) ))
+          }
+        }
+
+      } else if (gene_set_method == 'gprofiler2') {
+        enrichment_files <- grep("gprofiler2", list.files(params$input_dir), value=T, fixed=T)
+        if (length(grep(".html", enrichment_files, fixed=T))) {
+          cat(paste0("\nThis section contains the results of the pathway analysis which was done with the R package gprofiler2. The plots show the -log10 adjusted p values of each pathway which was found to be enriched for differential genes (if necessary determined by converting feature IDs to gene IDs); if possible, pathways are grouped by their source database. The tables below give additional info for each pathway; the differential fraction is the number of differential genes in a pathway divided by that pathway's size, i.e. the number of genes annotated for the pathway.",
+          ifelse(params$gprofiler2_significant, paste0(" Enrichment was only considered if significant, i.e. adjusted p-value <= ", params$gprofiler2_max_qval, "."), "Enrichment was also considered if not significant."), "\n"))
+
+          for (html in rev(grep(".html", enrichment_files, value=T, fixed=T))) {
+            contrast <- unlist(strsplit(html, "gprofiler2.", fixed=T))[2]
+            contrast <- unlist(strsplit(contrast, ".gostplot", fixed=T))[1]
+            if (! file.exists(file.path(params$input_dir, html))){
+              stop(paste("gprofiler2 gost plot", file.path(params$input_dir, html), "does not exist"))
+            }
+
+            cat(paste0("\n##### ", contrast, "\n"))
+            cat(paste0('<embed type="text/html" src="', file.path(params$input_dir, html), '" width="800" height="400">'))
```

Comment:
I'm not convinced by embedding random HTML in here, it could get messy and we're putting a lot of trust in the people who make those HTML files not to make them huge and unweildy. 

I could have done that for GSEA, but intentionally kept things to the summary tables and I think we should do the same for this. The user can go and refer to the individual HTML reports themselves.